gobblin-config-management/src/main/java/gobblin/config/configstore/impl/BaseHdfsConfigStore.java

```diff
@@ -0,0 +1,214 @@
+package gobblin.config.configstore.impl;
+
+import gobblin.config.configstore.ConfigStore;
+import gobblin.config.configstore.VersionComparator;
+import gobblin.config.utils.PathUtils;
+
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FSDataInputStream;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.apache.log4j.Logger;
+
+import com.google.common.io.Closer;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+
+/**
+ * The BaseHdfsConfigStore implements basic functions from ConfigStore which is related to HDFS
+ * @author mitu
+ *
+ */
+public abstract class BaseHdfsConfigStore implements ConfigStore {
```

Comment:
ETLHdfsConfigStore implements ConfigStoreWithStableVersion. Not sure BaseHdfs should implements ConfigStoreWithStableVersion